### PR TITLE
CI build

### PR DESCRIPTION
**Pull request by g4s8**
#2266 - configured CI build with GitHub actions

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#2267](https://github.com/zerocracy/farm/issues/2267) is assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/ed921d2f-1b12-4f83-91ab-30dba3012541); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Maven Build
+"on": [push, pull_request]
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [windows-latest, ubuntu-latest]
```
Comment (victornoel) [inline .github/workflows/main.yml:8]: @g4s8 does it really make sense to build it on windows too? is it expected to be deployed on windows?

**Comment by victornoel** [comment]
@g4s8 ping :)

File: .github/workflows/main.yml (hunk 1 above)
Comment (g4s8) [reply .github/workflows/main.yml:8]: @victornoel some people may use Windows for development and we need to be shure that the tests can be passed on this platform too

**Comment by g4s8** [comment]
@victornoel sorry for delay, I've answered your comment

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - uses: actions/cache@v1
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
```
Comment (victornoel) [inline .github/workflows/main.yml:18]: @g4s8 I think the repository is independent of whatever is in the file (contrary to say, a node_module with npm), so it is not needed to use a cache key based on the pom file, it should be enough to just have one dependent on the os.

File: .github/workflows/main.yml (hunk 1 above)
Comment (victornoel) [reply .github/workflows/main.yml:8]: @g4s8 ah indeed yes!

**Comment by victornoel** [comment]
@g4s8 I see you changed the cache key, do we still need the restore-key then?

**Comment by g4s8** [comment]
@victornoel thanks

**Comment by victornoel** [comment]
@g4s8 actually I believe restore key is useless here because the prefix is equal to the full key, so it will never be used theoretically :) anyway, it's not critical

**Comment by victornoel** [comment]
@g4s8 can you tell 0crat to close this job? It's stuck as open in my agenda

**Comment by victornoel** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/zerocracy/farm/issues/2267#issuecomment-619370910))

@victornoel This is what I know about this job in C3NDPUA8L, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#2267](https://github.com/zerocracy/farm/issues/2267) is [in scope](http://www.zerocracy.com/policy.html#14) for 8days
  * The role is `REV`
  * The job is assigned to @victornoel[/z](https://www.0crat.com/u/victornoel) for [8days](http://www.zerocracy.com/policy.html#8)
  * There is no monetary reward attached, it's a [free](http://www.zerocracy.com/policy.html#2) job
  * Some equity will be [vested](http://www.zerocracy.com/policy.html#37) on completion
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @g4s8[/z](https://www.0crat.com/u/g4s8): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/C3NDPUA8L?q=%7Bjob%3A%27gh%3Azerocracy%2Ffarm%232267%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by victornoel** [comment]
@g4s8 ping

**Comment by g4s8** [comment]
@0crat out

**Comment by 0crat** [comment]
Code review was too long (10 days), architects (@g4s8) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @victornoel[/z](https://www.0crat.com/u/victornoel), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/zerocracy/farm/issues/2267#issuecomment-620480557))

@g4s8 There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@470ff8ad-1642-4d34-9f48-355654d6ab84, thread: PQ-C3NDPUA8L
com.zerocracy.farm.reactive.StkRuntime[76] java.lang.IllegalStateException: org.codehaus.groovy.runtime.InvokerInvocationException in com.zerocracy.stk.pm.cost.vesting.transfer_shares
org.codehaus.groovy.reflection.CachedMethod[105] org.codehaus.groovy.runtime.InvokerInvocationException: java.lang.OutOfMemoryError: unable to create new native thread
java.lang.Thread[-2] java.lang.OutOfMemoryError: unable to create new native thread
```

[0.53.12](https://github.com/zerocracy/farm): CID: [3fc870a3-ea93-4941-a660-927ffc9616a0](https://www.0crat.com/footprint/C3NDPUA8L/3fc870a3-ea93-4941-a660-927ffc9616a0), Type: "Transfer shares"

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @victornoel[/z](https://www.0crat.com/u/victornoel)

**Pull request merged**